pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1480,6 +1480,16 @@ private boolean forceLog(BrokerResponse brokerResponse, long totalTimeMs) {
   @VisibleForTesting
   static void setOptions(PinotQuery pinotQuery, long requestId, String query, JsonNode jsonRequest) {
     Map<String, String> queryOptions = new HashMap<>();
+    if (jsonRequest.has(Broker.Request.DEBUG_OPTIONS)) {
+      Map<String, String> debugOptions = getOptionsFromJson(jsonRequest, Broker.Request.DEBUG_OPTIONS);
+      if (!debugOptions.isEmpty()) {
+        LOGGER.debug("Debug options are set to: {} for request {}: {}", debugOptions, requestId, query);
+        pinotQuery.setDebugOptions(debugOptions);
```

Comment:
We should not setDebugOptions at all IMO. Just putting it in query options ensures backward compatibility. We don't want other users/devs to rely on debug options for anything. Also `@Deprecated` annotation on `setDebugOptions`

Reply:
We do need to set it here for backward compatibility because the server might have not been upgraded to the newer version and is still expecting the option from the debug options. I didn't annotate `PinotQuery.setDebugOptions()` because it is auto-generated thrift class. Let me add some comments here to remove this after releasing `0.11.0`